### PR TITLE
{chem}[intel/2020a] CP2K v7.1, libxsmm v1.16.1, libxc-4.3.4

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

requires https://github.com/easybuilders/easybuild-easyblocks/pull/2069 + #11099 (Libint)

### CODE REVIEW

File: easybuild/easyconfigs/c/CP2K/CP2K-7.1-intel-2019b.eb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name = 'CP2K'
+version = '7.1'
+
+homepage = 'http://www.cp2k.org/'
```
Comment (Micket) [inline easybuild/easyconfigs/c/CP2K/CP2K-7.1-intel-2019b.eb:4]:
```suggestion
homepage = 'https://www.cp2k.org/'
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3402.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/c0f9249cbf3777728df171796306fcfa for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3103.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/3121b823b92a43ddd8341c216ae224b4 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
generoso-x-1 - Linux centos linux 8.2.2004, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8
See https://gist.github.com/cdb314fd8bd56a6dde22559184cf95ce for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2612.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/39202e9bb4c1c006520942df9f367f77 for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 4 out of 6 (3 easyconfigs in this PR)
node254.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2683 v4 @ 2.10GHz (broadwell), Python 2.7.5
See https://gist.github.com/5f5e49c4e0a5dafd00213c3e6e329d10 for a full test report.

**Comment by boegel** [comment]
@smoors A bunch of failing tests for `libxc`, several with timeouts: `99% tests passed, 12 tests failed out of 20302`...

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
node368.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/36fdd851bdf334eef8b3759eff3ad6e9 for a full test report.

**Comment by smoors** [comment]
@boegel the failed tests are due to the timeouts as you can see:
```
The following tests FAILED:
1917 - lda_c_xalpha-BrOH+-unpol-0-run (Timeout)
1918 - lda_c_xalpha-BrOH+-unpol-0-check (Failed)
3441 - gga_x_cap-Li-unpol-0-run (Timeout)
3442 - gga_x_cap-Li-unpol-0-check (Failed)
4741 - gga_x_mb88-BrOH-pol-0-run (Timeout)
4742 - gga_x_mb88-BrOH-pol-0-check (Failed)
5339 - gga_x_pbe_sol-Li-pol-1-run (Timeout)
5340 - gga_x_pbe_sol-Li-pol-1-check (Failed)
13709 - mgga_x_vt84-BrOH+-pol-1-run (Timeout)
13710 - mgga_x_vt84-BrOH+-pol-1-check (Failed)
18839 - hyb_mgga_x_m05_2x-H-unpol-0-run (Timeout)
18840 - hyb_mgga_x_m05_2x-H-unpol-0-check (Failed)
```

with the following patch all tests pass for me:
```
diff -ur libxc-4.3.4.orig/testsuite/CMakeLists.txt libxc-4.3.4/testsuite/CMakeLists.txt
--- libxc-4.3.4.orig/testsuite/CMakeLists.txt	2019-01-25 15:47:40.000000000 +0100
+++ libxc-4.3.4/testsuite/CMakeLists.txt	2020-08-13 23:34:08.091740000 +0200
@@ -42,7 +42,7 @@
         set(test "${func}-${system}-${spin}-${order}")
         add_test(NAME ${test}-run
                  COMMAND xc-regression ${func} ${nspin} ${order} ${CMAKE_CURRENT_SOURCE_DIR}/input/${system} ${CMAKE_CURRENT_BINARY_DIR}/workdir/${_sdir}/${refname})
-        set_tests_properties(${test}-run PROPERTIES TIMEOUT 1)
+        set_tests_properties(${test}-run PROPERTIES TIMEOUT 20)
 
         # checke the tests values against the reference values
         add_test(NAME ${test}-check
```

I guess it does not hurt to add this patch?

**Comment by smoors** [comment]
the timeouts themselves could be due to our shared storage being temporarily unresponsive.

**Comment by boegel** [comment]
@pouillon Thoughts on increased the timeout in the `libxc` tests to be a bit more forgiving w.r.t. slow filesystems?

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
sms - Linux centos linux 7.6.1810, x86_64, AMD EPYC 7601 32-Core Processor (zen), Python 2.7.5
See https://gist.github.com/bd78e9709417eaa4c1f536bfc358e565 for a full test report.

**Comment by boegel** [comment]
@migueldiascosta failed due to existing lock dir :)

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
sms - Linux centos linux 7.6.1810, x86_64, AMD EPYC 7601 32-Core Processor (zen), Python 2.7.5
See https://gist.github.com/3810f7b9d1d24553bd8c6e5a2113650a for a full test report.

**Comment by boegel** [comment]
@migueldiascosta What's the tally on the regression test on your AMD Epyc system? Look for `Summary` in the CP2K log...

Here it is for my test installations, same results on Haswell/Skylake/Cascade Lake:

```
--------------------------------- Summary --------------------------------
Number of FAILED  tests 1
Number of WRONG   tests 8
Number of CORRECT tests 3253
Number of NEW     tests 8
Total number of   tests 3270
--------------------------------------------------------------------------
Number of memory  leaks 0
```

(some CP2K tests always fail, and it's difficult to tell which ones are fine to ignore, but the fact that the results are identical across systems is encouraging)

**Comment by migueldiascosta** [comment]
@boegel same results on AMD Epyc (so at least a test checking the number of failed/wrong tests would pass :p )

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
node254.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2683 v4 @ 2.10GHz (broadwell), Python 2.7.5
See https://gist.github.com/5acce1741a1c2020a8b333d905bcf0c0 for a full test report.

**Comment by smoors** [comment]
I'll submit a test for CP2K only (after installing libxc with the above patch applied).

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node252.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz (broadwell), Python 2.7.5
See https://gist.github.com/b4d21e8a92e8536e42a841a1b91c53b6 for a full test report.

**Comment by smoors** [comment]
I get exactly the same failed/wrong tests:
```
Number of FAILED  tests 1
Number of WRONG   tests 8
Number of CORRECT tests 3253
Number of NEW     tests 8
Total number of   tests 3270
--------------------------------------------------------------------------
Number of memory  leaks 0
```

**Comment by boegel** [comment]
@smoors Shall we apply that `libxc` patch in a different PR?

**Comment by smoors** [comment]
> @smoors Shall we apply that `libxc` patch in a different PR?

fine, I'll write the PR.

**Comment by smoors** [comment]
Going in, thanks @boegel!

**Pull request merged**